### ISSUE TITLE
Publishing rate is not accurate when using a variable rate flag

### ISSUE THREAD

**Issue by mkuratczyk** [rabbitmq/rabbitmq-perf-test#385]
When using variable rate, perf-test seems to consistently send a bit more than the rate requested. Compare this:
```
$ perf-test -x 5 -y 0 -c 1 -r 10
...
id: test-115039-259, time: 1.021s, sent: 45 msg/s, confirmed: 44 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 210/336/400/513/1792 µs
id: test-115039-259, time: 2.021s, sent: 50 msg/s, confirmed: 50 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 160/299/362/471/580 µs
id: test-115039-259, time: 3.021s, sent: 50 msg/s, confirmed: 50 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 215/358/410/510/763 µs
id: test-115039-259, time: 4.021s, sent: 50 msg/s, confirmed: 50 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 221/369/406/450/535 µs
id: test-115039-259, time: 5.021s, sent: 50 msg/s, confirmed: 50 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 194/346/406/534/566 µs
id: test-115039-259, time: 6.021s, sent: 50 msg/s, confirmed: 50 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 170/276/331/429/493 µs
```

With this:
```
$ perf-test -x 5 -y 0 -c 1 -vr 10:100
...
id: test-115050-684, time: 1.019s, sent: 45 msg/s, confirmed: 44 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 262/392/460/1819/1929 µs
id: test-115050-684, time: 2.019s, sent: 55 msg/s, confirmed: 56 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 232/333/421/643/674 µs
id: test-115050-684, time: 3.019s, sent: 55 msg/s, confirmed: 55 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 164/272/317/463/568 µs
id: test-115050-684, time: 4.019s, sent: 56 msg/s, confirmed: 55 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 164/271/305/448/549 µs
id: test-115050-684, time: 5.019s, sent: 54 msg/s, confirmed: 55 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 158/268/337/450/503 µs
id: test-115050-684, time: 6.020s, sent: 54 msg/s, confirmed: 53 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 130/233/273/412/472 µs
```

I've tried multiple times on two different machines and with different queue types and it seems consistent - 50 when using `-r` but 54-56 msg/s when using `-vr`.

**Comment by mkuratczyk** [comment]
It seems to scale with the number of publishers but not with the message rate:

```
$ perf-test -x 50 -y 0 -c 1 -vr 10:100
...
id: test-121559-768, time: 1.082s, sent: 370 msg/s, confirmed: 369 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 130/232/324/812/2498 µs
id: test-121559-768, time: 2.082s, sent: 551 msg/s, confirmed: 551 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 107/190/216/280/342 µs
id: test-121559-768, time: 3.082s, sent: 549 msg/s, confirmed: 549 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 117/185/219/363/481 µs
id: test-121559-768, time: 4.082s, sent: 550 msg/s, confirmed: 550 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 102/177/206/265/367 µs
id: test-121559-768, time: 5.082s, sent: 550 msg/s, confirmed: 550 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 101/187/222/311/514 µs
id: test-121559-768, time: 6.082s, sent: 550 msg/s, confirmed: 550 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 96/166/194/257/372 µs
id: test-121559-768, time: 7.083s, sent: 549 msg/s, confirmed: 549 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 78/162/186/251/331 µs
```

With a few faster publishers, the problem pretty much disappears
```
$ perf-test -x 5 -y 0 -c 1 -vr 1000:100
...
id: test-121809-664, time: 1.000s, sent: 4375 msg/s, confirmed: 4375 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 79/139/168/239/369 µs
id: test-121809-664, time: 2.000s, sent: 5007 msg/s, confirmed: 5007 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 69/112/127/159/196 µs
id: test-121809-664, time: 3.000s, sent: 5005 msg/s, confirmed: 5004 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 71/111/129/164/195 µs
id: test-121809-664, time: 4.000s, sent: 5003 msg/s, confirmed: 5003 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 67/109/131/166/194 µs
id: test-121809-664, time: 5.000s, sent: 5003 msg/s, confirmed: 5002 msg/s, nacked: 0 msg/s, min/median/75th/95th/99th confirm latency: 72/108/128/163/203 µs
```

**Comment by acogoluegnes** [comment]
Fixed in #405.
